tests/lapi/bitop_band_test.lua

```diff
@@ -0,0 +1,44 @@
+--[[
+SPDX-License-Identifier: ISC
+Copyright (c) 2023-2025, Sergey Bronnikov.
+
+Wrong code generation for constants in bitwise operations,
+https://github.com/lua/lua/commit/c764ca71a639f5585b5f466bea25dc42b855a4b0
+
+Inconsistent behaviour of bit ops in DUALNUM mode,
+https://github.com/LuaJIT/LuaJIT/issues/1273
+
+Synopsis: bit32.band (...)
+]]
+
+local luzer = require("luzer")
+local test_lib = require("lib")
+
+local band
+if test_lib.version() == "LuaJIT" then
+    local bitop = require("bit")
+    band = bitop.band
+else
+    band = test_lib.bitwise_op("&")
+end
+
+local function TestOneInput(buf)
+    local fdp = luzer.FuzzedDataProvider(buf)
+    local x = test_lib.random_number(fdp)
+    local y = test_lib.random_number(fdp)
+    local z = test_lib.random_number(fdp)
+
+    assert(band(x, band(y, z)) ==
+           band(band(x, y), z), "x & (y & z) ~= (x & y) & z")
+    assert(band(x, 0) == 0, "x & 0 ~= 0, where x = " .. x)
+
+    -- Broken?
+    -- assert(band(x, 0xffff) == x, "x & 0xFFFF ~= x")
+    -- assert(band(x, y) == band(y, z), "x & y ~= y & x")
```

Comment:
Why do we check `band(x,y) == band(y,z)`? Should it be `band(x,y) == band(y,x)`?

Reply:
Fixed.